repository_service_tuf/cli/admin/metadata.py

```diff
@@ -0,0 +1,626 @@
+                break
+
+
+def _keys_additions(current_root: RootInfo):
+    root_threshold = current_root.threshold
+    console.print(
+        f"You need to have at least [cyan]{root_threshold}[/] signing keys."
+    )
+    while True:
+        keys_table = _create_keys_table(
+            current_root.signing_keys,
+            is_online_key_table=False,
+            is_minimal=False,
+        )
+        console.print("\nHere are the current root signing keys:")
+        console.print(keys_table)
+        response = prompt.Confirm.ask("\nDo you want to add a new key?")
+        if response is False:
+            keys_amount = len(current_root.signing_keys)
+            if keys_amount < root_threshold:
+                remaining = root_threshold - keys_amount
+                console.print(
+                    f"You need to add an additional {remaining} signing keys"
+                )
+                abort = prompt.Confirm.ask(
+                    "Do you want to abort the root metadata update",
+                )
+                if abort:
+                    raise click.ClickException(
+                        "Not enough keys to fulfill the threshold requirement"
+                    )
+                else:
+                    continue
+            else:
+                break
+
+        root_key: RSTUFKey = _get_key(Root.type)
+        if root_key.error:
+            console.print(f":cross_mark: [red]Failed[/]: {root_key.error}")
+            continue
+
+        if root_key == current_root.online_key:
+            console.print(
+                ":cross_mark: [red]Failed[/]: This is the current online key. "
+                "Cannot be added"
+            )
+            continue
+
+        if current_root.is_keyid_used(root_key.key["keyid"]):
+            console.print(":cross_mark: [red]Failed[/]: Key is already used")
+            continue
+
+        root_key.name = prompt.Prompt.ask(
+            "[Optional] Give a [green]name/tag[/] to the key"
+        )
+
+        current_root.add_key(root_key)
+
+
+def _get_positive_int_input(msg: str, input_name: str, default: Any) -> int:
+    input: int = 0
+    while input < 1:
+        input = prompt.IntPrompt.ask(msg, default=default, show_default=True)
+        if input < 1:
+            console.print(f"{input_name} must be at least 1")
+            continue
+
+    return input
+
+
+def _modify_expiration(current_root: RootInfo):
+    console.print(markdown.Markdown(EXPIRY_CHANGES_MSG), width=100)
+    console.print("\n")
+    while True:
+        console.print(
+            f"Current root expiration: [cyan]{current_root.expiration_str}[/]",
+            highlight=False,  # disable built-in rich highlight
+        )
+        change = prompt.Confirm.ask(
+            "Do you want to extend the [cyan]root's expiration[/]?"
+        )
+        if change is False:
+            if current_root.expiration < (datetime.now() + timedelta(days=1)):
```

Comment:
It's not very user-friendly to first give the user a choice and then check if they even have a choice. If they have to change it anyway, if it's expired, you can ask them to change right away.